juju/model.py

```diff
@@ -543,6 +543,7 @@ def connection(self):
                 raise ValueError('Authentication parameters are required '
                                  'if model_name not given')
             await self._connector.connect(**kwargs)
+        self.uuid = self.connection().connect_params()['uuid']
```

Comment:
This line is causing the unit test failures:

```
Traceback (most recent call last):
  File "/home/travis/build/juju/python-libjuju/tests/unit/test_model.py", line 212, in test_with_endpoint_and_uuid_with_bakery
    bakery_client='bakery')
  File "/home/travis/build/juju/python-libjuju/juju/model.py", line 546, in connect
    self.uuid = self.connection().connect_params()['uuid']
  File "/home/travis/build/juju/python-libjuju/juju/model.py", line 444, in connection
    return self._connector.connection()
  File "/home/travis/build/juju/python-libjuju/juju/client/connector.py", line 49, in connection
    raise NoConnectionException('not connected')
juju.client.connector.NoConnectionException: not connected
```

The `self._connector.connect()` call gets mocked out to a no-op, but this call isn't intercepted.